src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs

```diff
@@ -234,13 +235,94 @@ public AndroidTzData()
+            //       ...
+            //       <id>Australia/Eucla</id>
+            //     </country>
+            //     <country ...>
+            //       ...
+            //       ...
+            //       ...
+            //     </country>
+            //   </countryzones>
+            // </timezones>
+            //
+            // Once the timezone cache is populated with the IDs, we reference tzlookup id tags and
+            // remove IDs and their associated information (byteoffsets and lengths) from the current
+            // AndroidTzData instance.
+            private void filterBackwardIDs(string lookupPath)
+            {
+                try
+                {
+                    using (StreamReader sr = File.OpenText(lookupPath))
+                    {
+                        var tzLookupIDs = new HashSet<string>();
+                        string? tzLookupLine;
+                        while (sr.Peek() >= 0)
+                        {
+                            if (!(tzLookupLine = sr.ReadLine())!.TrimStart().StartsWith("<id"))
+                                continue;
+
+                            int idStart = tzLookupLine.IndexOf('>')+1;
+                            int idLength = tzLookupLine.IndexOf("</")-idStart;
+                            if (idStart == 0 || idLength < 0)
+                            {
+                                // Either the start tag <id ... > or the end tag </id> are not found
+                                continue;
+                            }
+                            string id = tzLookupLine.Substring(idStart, idLength);
+                            tzLookupIDs.Add(id);
+                        }
+                        var cleanIDs = new List<string>();
+                        var cleanOffsets = new List<int>();
+                        var cleanLengths = new List<int>();
+                        for (int i = 0; i < _ids.GetLength(0); ++i)
+                        {
+                            if (tzLookupIDs.Contains(_ids[i]))
+                            {
+                                cleanIDs.Add(_ids[i]);
+                                cleanOffsets.Add(_byteOffsets[i]);
+                                cleanLengths.Add(_lengths[i]);
+                            }
+                        }
+                        _ids = cleanIDs.ToArray();
+                        _byteOffsets = cleanOffsets.ToArray();
+                        _lengths = cleanLengths.ToArray();
+                    }
+                }
+                catch (FileNotFoundException)
+                {
+                    // Some versions of Android will not have tzlookup.xml, so just warn that there may be backward timezone IDs
+                }
+                catch (IOException)
+                {
+                    throw new InvalidOperationException(SR.Format(SR.InvalidOperation_TimeZoneLookupParsing, lookupPath));
+                }
+                catch (OutOfMemoryException)
```

Comment:
Just curious why out of memory?

Reply:
I just looked through all the APIs being used in the method and looked at what possible exceptions could be thrown. We can omit this exception if its not necessary.

Reply:
We are now catching all exceptions per https://github.com/dotnet/runtime/pull/64028#discussion_r789011873